lib/VeresOneClient.js

```diff
@@ -44,18 +45,36 @@ class VeresOneClient {
    * it's likely a key id, so just return the subgraph, not the full doc.
    *
    * @param did {string} DID uri (possibly with hash fragment)
+   * @param forceConstruct {boolean} Forces deterministic construction of
+   *   DID Document from cryptonym.
    *
    * @returns {Promise<object>} Resolves to DID Document Fetch Result
    */
-  async get({did}) {
+  async get({did, forceConstruct = false}) {
     if(!did) {
       throw new Error('Invalid or missing DID URI.');
     }
     const [docUri, hashFragment] = did.split('#');
-
-    const result = await this.ledger.getRecord({id: docUri});
-
-    const didDoc = result.record;
+    const isNym = (did.startsWith('did:v1:nym:') ||
+      did.startsWith('did:v1:test:nym:'));
+    let result = {};
+    let didDoc;
+
+    // FIXME - remove, replace with cache
+    if(isNym && forceConstruct) {
+      didDoc = (await VeresOneDidDoc.fromNym({did: docUri})).doc;
+    } else {
+      try {
+        result = await this.ledger.getRecord({id: docUri});
+        didDoc = result.record;
+      } catch(error) {
+        if(error.name === 'NotFoundError' && isNym) {
```

Comment:
I believe the `isNym` here is not required because it's already tested on L64.

Reply:
It's possible that `forceConstruct` could have been false and you'd need to test again.